pandas/core/construction.py

```diff
@@ -807,6 +807,8 @@ def _try_cast(
         )
 
     elif dtype.kind in "mM":
+        if is_ndarray and arr.ndim == 2 and arr.shape[1] == 1:
+            return maybe_cast_to_datetime(arr.ravel(), dtype).reshape(arr.shape)
```

Comment:
instead of `arr.ravel()` (which can make a copy), can you do `arr[:, 0]`?

Reply:
I have added a comment and applied this change.